crates/timestamp/Cargo.toml

```diff
@@ -0,0 +1,40 @@
+[package]
```

Comment:
I've mostly copied the kitsune_p2p_timestamp crate but I tidied up the Cargo.toml quite a bit. There were several unused dependencies and a mixture of feature names and "dependency features" being used in the code. We're now using `dep:` so the feature logic is a bit more consistent and hopefully easier to follow.